bin/faucet/src/main.rs

```diff
@@ -229,3 +196,139 @@ fn long_version() -> LongVersion {
         debug: option_env!("VERGEN_CARGO_DEBUG").unwrap_or_default(),
     }
 }
+
+async fn serve_faucet(config: FaucetConfig) -> anyhow::Result<()> {
+    let faucet_state = FaucetState::new(config.clone()).await?;
+
+    info!(target: COMPONENT, %config, "Initializing server");
+
+    let app = create_router(faucet_state);
+
+    let socket_addr = config
+        .endpoint
+        .socket_addrs(|| None)?
+        .into_iter()
+        .next()
+        .with_context(|| format!("no sockets available on {}", config.endpoint))?;
+    let listener = TcpListener::bind(socket_addr).await.context("failed to bind TCP listener")?;
+
+    info!(target: COMPONENT, endpoint = %config.endpoint, "Server started");
+
+    axum::serve(listener, app).await.unwrap();
+    Ok(())
+}
+
+fn create_router(faucet_state: FaucetState) -> Router {
+    Router::new()
+        .route("/", get(get_index_html))
+        .route("/index.js", get(get_index_js))
+        .route("/index.css", get(get_index_css))
+        .route("/background.png", get(get_background))
+        .route("/favicon.ico", get(get_favicon))
+        .route("/get_metadata", get(get_metadata))
+        .route("/get_tokens", post(get_tokens))
+        .layer(
+            ServiceBuilder::new()
+                .layer(TraceLayer::new_for_http())
+                .layer(SetResponseHeaderLayer::if_not_present(
+                    http::header::CACHE_CONTROL,
+                    HeaderValue::from_static("no-cache"),
+                ))
+                .layer(
+                    CorsLayer::new()
+                        .allow_origin(tower_http::cors::Any)
+                        .allow_methods(tower_http::cors::Any),
+                ),
+        )
+        .with_state(faucet_state)
+}
+
+#[cfg(test)]
+mod test {
+    use std::{
+        io::{BufRead, BufReader},
+        process::{Command, Stdio},
+        str::FromStr,
+    };
+
+    use fantoccini::ClientBuilder;
+    use serde_json::{json, Map};
+    use url::Url;
+
+    use crate::{config::FaucetConfig, serve_faucet, stub_rpc_api::serve_stub};
+
+    /// This test starts a stub node, a faucet connected to the stub node, and a chromedriver
+    /// to test the faucet website. It then loads the website and checks that all the requests
+    /// made return status 200.
+    #[tokio::test]
+    async fn test_website() {
+        let stub_node_url = Url::from_str("http://localhost:50051").unwrap();
+
+        // Start the stub node
+        tokio::spawn({
+            let stub_node_url = stub_node_url.clone();
+            async move { serve_stub(&stub_node_url).await.unwrap() }
+        });
+        // Start the faucet connected to the stub
+        let config = FaucetConfig {
+            node_url: stub_node_url,
+            faucet_account_path: "test_data/faucet.mac".into(),
```

Comment:
Would it be possible to generate this as part of the test instead?

I mainly ask because I dislike random binary files floating around the codebase.

This does bring up a slight refactoring though - if we factor out the `match &cli.command` from `main.rs` into a separate function, then we could:

- Revert the `serve_faucet` changes (move them back to `Command::start`. This isn't overly important, just a smaller diff.
- Run the full faucet init, account creation and start commands in order.

This would also require using `tempfs` to host the files.

I rather like this idea. Though maybe it would be best to merge this PR and upgrade it in another PR.

@TomasArrachea WDYT? cc @igamigo